packages/plugin-table/src/TableDomParser.ts

```diff
@@ -68,14 +68,28 @@ export class TableDomParser extends AbstractParser<Node> {
      * @param domTable
      */
     _getTableDimensions(domTable: HTMLTableElement): [number, number] {
-        const domRows = Array.from(domTable.querySelectorAll('tr'));
+        const domRows = Array.from(domTable.querySelectorAll('tr'))
+        const domTableRows = domRows.filter(row => this._domTableAncestor(row) === domTable);
         let columnCount = 0;
-        if (domRows.length) {
-            for (const domChild of domRows[0].querySelectorAll('td, th')) {
+        if (domTableRows.length) {
+            const domCells = Array.from(domTableRows[0].querySelectorAll('td, th'));
+            const domTableCells = domCells.filter(cell => this._domTableAncestor(cell) === domTable);
```

Comment:
The dom have `closest`, doesn't it ? I suggest trying with `closest` in order to reove this auxiliary function.

Reply:
Man good job, I didn't know this existed, I thought that was only in jQuery...